src/main/java/pl/javastart/task/phone/MixPhoneContract.java

```diff
@@ -0,0 +1,92 @@
+package pl.javastart.task.phone;
+
+import java.util.Objects;
+
+public class MixPhoneContract extends CardPhoneContract {
+    int smsAmountLeft;
+    int mmsAmountLeft;
+    int callMinutesAmountLeft;
+
+    public MixPhoneContract(double accountState, double smsCost, double mmsCost, double callCostPerMinute, 
+                            int startSmsAmount, int startMmsAmount, int startCallMinutesAmount) {
+        super(accountState, smsCost, mmsCost, callCostPerMinute);
+        this.smsAmountLeft = startSmsAmount;
+        this.mmsAmountLeft = startMmsAmount;
+        this.callMinutesAmountLeft = startCallMinutesAmount;
+    }
+
+    @Override
+    boolean performSmsSending() {
+        if (smsAmountLeft > 0) {
+            sentSms++;
+            smsAmountLeft--;
+            return true;
+        } else {
+            return super.performSmsSending();
+        }
+    }
+
+    @Override
+    int performCalling(int seconds) {
+        int callMinutesAmountLeftInSeconds = callMinutesAmountLeft * SECONDS_IN_MINUTE;
+        if (callMinutesAmountLeftInSeconds >= seconds) {
+            callMinutesAmountLeft -= seconds / SECONDS_IN_MINUTE;
+            callSeconds += seconds;
+            return seconds;
+        } else {
+            callMinutesAmountLeft = 0;
+            callSeconds += callMinutesAmountLeftInSeconds;
+            int phoneCallInSeconds = super.performCalling(seconds - callMinutesAmountLeftInSeconds);
+            return phoneCallInSeconds + callMinutesAmountLeftInSeconds;
+        }
+    }
```

Comment:
Dla takiego przypadku
```
Phone phone3 = new Phone(new MixPhoneContract(2, 0.1, 0.3, 1, 1, 1, 1));
phone3.call(180);
```
jest błędny wynik. Najpierw powinniśmy zużyć minuty w pakiecie, a później dodatkowo część środków z konta. U Ciebie rozmowa kończy się po 100 sekundach, a na koncie pozostaje 2zł

Reply:
Wygląda na to, że można dzwonić u Ciebie bez końca i środków nie ubywa:
```
Phone phone3 = new Phone(new MixPhoneContract(0.1, 0.1, 0.3, 1, 1, 1, 1));
System.out.println("Telefon mix");
phone3.call(40);
phone3.printAccountState();
phone3.call(40);
phone3.printAccountState();
phone3.call(40);
phone3.printAccountState();
```